spec/models/preserved_copy_spec.rb

```diff
@@ -320,49 +321,55 @@
   describe '#create_zipped_moab_versions!' do
     let(:pc_version) { 3 }
     let(:archive_ep) { ZipEndpoint.find_by!(endpoint_name: 'mock_archive1') }
-    let(:new_archive_ep) { create(:zip_endpoint, endpoint_name: 'mock_archive2') }
-
-    it "creates pres copies that don't yet exist for the given version, but should" do
-      expect { pc.create_zipped_moab_versions!(pc_version) }.to change {
-        ZipEndpoint.which_need_archive_copy(druid, pc_version).to_a
-      }.from([archive_ep]).to([])
+    let(:zmvs_by_druid) { ZippedMoabVersion.by_druid(druid) }
 
-      expect(ZippedMoabVersion.by_druid(druid).count).to eq 1
+    before { pc.zipped_moab_versions.destroy_all } # undo auto-spawned rows from callback
 
-      expect { pc.create_zipped_moab_versions!(pc_version - 1) }.to change {
-        ZipEndpoint.which_need_archive_copy(druid, pc_version - 1).to_a
-      }.from([archive_ep]).to([])
-      expect(ZippedMoabVersion.by_druid(druid).count).to eq 2
+    it "creates ZMVs that don't yet exist for expected versions, but should" do
+      expect { pc.create_zipped_moab_versions! }.to change {
+        ZipEndpoint.which_need_archive_copy(druid, pc_version).to_a
+      }.from([archive_ep]).to([]).and change {
+        zmvs_by_druid.where(version: pc_version).count
+      }.from(0).to(1)
 
-      expect(ZippedMoabVersion.by_druid(druid).where(version: 1).count).to eq 0
+      expect(zmvs_by_druid.pluck(:version).sort).to eq [1, 2, 3]
     end
 
-    it 'creates the pres copies so that they start with unreplicated status' do
-      expect(pc.create_zipped_moab_versions!(pc_version).all?(&:unreplicated?)).to be true
+    it 'creates ZMVs so that they start with unreplicated status' do
+      expect(pc.create_zipped_moab_versions!.all?(&:unreplicated?)).to be true
     end
 
-    it "creates pres copies that don't yet exist for the given moab_storage_root, but should" do
-      expect { pc.create_zipped_moab_versions!(pc_version) }.to change {
+    it "creates ZMVs that don't yet exist for new endpoint, but should" do
+      expect { pc.create_zipped_moab_versions! }.to change {
         ZipEndpoint.which_need_archive_copy(druid, pc_version).to_a
-      }.from([archive_ep]).to([])
-      expect(ZippedMoabVersion.by_druid(druid).where(version: pc_version).count).to eq 1
+      }.from([archive_ep]).to([]).and change {
```

Comment:
neat, i wasn't aware you could do these compound expectations for the `change` matcher.  very handy.